website/docs/faqs/Warehouse/db-connection-dbt-compile.md

```diff
@@ -0,0 +1,17 @@
+---
+title: Why does dbt compile need a data platform connection?
+description: "`dbt compile` needs a data platform connection to perform introspective queries and generate the compiled SQL"
+sidebar_label: "Why does dbt compile need a data platform connection?"
+id: db-connection-dbt-compile
+---
+
+The [`dbt compile`](reference/commands/compile) command is a really important step in the dbt workflow. This is because it transforms the raw SQL in dbt models and tests into compiled SQL, which is appropriate to run on a specific data platform. The compiled SQL includes SQL queries that are generated by dbt, such as query parts for incremental models, `ref` macros, and other [dbt-specific syntax](reference/node-selection/syntax).
+
+To generate the compiled SQL for many models, dbt needs to run introspective queries, (when dbt needs to run SQL in order to pull data back and do something with it) against the data platform. These introspective queries include populating the [relation cache](/guides/advanced/creating-new-materializations#update-the-relation-cache), resolving [macros](/docs/build/jinja-macros#macros), and checking if models are [incremental](/docs/build/incremental-models). Without a database connection, dbt can't perform these introspective queries and won't be able to generate the compiled SQL needed for the next steps in the dbt workflow.
```

Comment:
Does "resolving macros" here derive from jerco's comment of "resolve things like `is_incremental()` and `ref()`"? I read it as resolving custom macros, but I don't know what macro resolution would mean here in that context.

If that's not what you meant then you'll have to tweak this suggestion a bit! 

```suggestion
To generate the compiled SQL for many models, dbt needs to run introspective queries, (when dbt needs to run SQL in order to pull data back and do something with it) against the data platform. These introspective queries include:
- populating the [relation cache](/guides/advanced/creating-new-materializations#update-the-relation-cache),
- resolving [`ref`s](/reference/dbt-jinja-functions/ref) when deferring to an earlier run,
- checking if models are [incremental](/docs/build/incremental-models),
- using package macros like [get_column_values](https://github.com/dbt-labs/dbt-utils#get_column_values-source) or [get_relations_by_pattern](https://github.com/dbt-labs/dbt-utils#get_relations_by_pattern-source).
Without a database connection, dbt can't generate the correct SQL to run.
```